src/js/components/PodSpecView.js

```diff
@@ -0,0 +1,172 @@
+import React from 'react';
+
+import DescriptionList from './DescriptionList';
+import Icon from './Icon';
+import PodSpec from '../structs/PodSpec';
+
+class PodSpecView extends React.Component {
+  getEnvironmentDetails() {
+    let environment = this.props.spec.getEnvironment();
+
+    if (Object.keys(environment).length === 0) {
+      return null;
+    }
+
+    let hash = Object.keys(environment).reduce(function (memo, key) {
+      let value = environment[key];
+
+      if (typeof value === 'object' && !Array.isArray(value) &&
+          value !== null) {
```

Comment:
I like that! 
There's an idea to extract this check into a utility function like `Utils.isObject` in a separate PR of course.